lib/pure/options.nim

```diff
@@ -70,31 +70,54 @@
 
 import typetraits
 
+type
+  SomePointer = ref | ptr | pointer
 
 type
   Option*[T] = object
     ## An optional type that stores its value and state separately in a boolean.
-    val: T
-    has: bool
+    when T is SomePointer:
+      val: T
+    else:
+      val: T
```

Comment:
Why don't you put just the `has` field in a `when T isnot SomePointer:` statement?